napari/_qt/qt_console.py

```diff
@@ -0,0 +1,84 @@
+from qtconsole.rich_jupyter_widget import RichJupyterWidget
+from qtconsole.inprocess import QtInProcessKernelManager
+from qtconsole.manager import QtKernelManager
+from qtconsole.client import QtKernelClient
+from IPython import get_ipython
+from IPython.terminal.interactiveshell import TerminalInteractiveShell
+from ipykernel.inprocess.ipkernel import InProcessInteractiveShell
+from ipykernel.zmqshell import ZMQInteractiveShell
+from ipykernel.connect import get_connection_file
+
+
+class QtConsole(RichJupyterWidget):
+    """Qt view for console.
+
+    Parameters
+    ----------
+    user_variables : dict
+        Dictionary of user variables to declare in console name space.
+
+    Attributes
+    ----------
+    kernel_client : qtconsole.inprocess.QtInProcessKernelClient,
+                    qtconsole.client.QtKernelClient, or None
+        Client for the kernel if it exists, None otherwise.
+    shell : ipykernel.inprocess.ipkernel.InProcessInteractiveShell,
+            ipykernel.zmqshell.ZMQInteractiveShell, or None.
+        Shell for the kernel if it exists, None otherwise.
+    """
+
+    def __init__(self, user_variables=None):
+        super().__init__()
+
+        # get current running instance or create new instance
+        shell = get_ipython()
+
+        if shell is None or type(shell) == InProcessInteractiveShell:
+            # If there is no currently running instance create an in-process
+            # kernel or if there is an old running InProcessInteractiveShell
+            # then just create a new one - necessary for our tests to pass
```

Comment:
have changed this - the problem comes when we have multiple viewers launched from the same python script - could be because of our tests, or could not be. Right now it doesn't work so well. I think we can bump fixing this to followup PRs as I don't think it is a top priority use case